app/views/session.py

```diff
@@ -35,15 +36,17 @@ def login():
     if session:
         session.clear()
 
-    logger.debug("attempting token authentication")
-    authenticator.jwt_login(request)
-    logger.debug("token authenticated - linking to session")
+    token = decrypt_token(request.args.get('token'))
+    jti_token = token.get('jti')
+    if jti_token is not None and not use_token(jti_token):
```

Comment:
Seems like the validation stuff is something the PyJWT should check for as it's in the JWT spec but not sure if it does. Not sure if us checking it's a valid uuid makes it anymore secure. If they chose to use some other globally unique uuid would that be bad (other than not implementing by the spec)